design-documents/media/catalog-images.md

```diff
@@ -0,0 +1,77 @@
+   * Might have Base CDN URL. This is similar to current Base Media URL and serves the same purpose, but might exist in case both sources of assets should be supported (Magento and CDN).
+
+## Scenarios
+
+### Backoffice Scenarios
+
+#### Assign an image to a product (using DAM integration)
+
+This is a future desired scenario, provided here for better understanding of the future picture.
+
+1. Admin opens product edit page
+2. Admin uses asset picker UI (provided as part of DAM integration) to select necessary image
+3. Admin clicks "Save"
+  * Image is linked to the product as provided by DAM
+  * Image path relative to DAM base URL is stored as image path
+  * Asset is assigned to the product in DAM
+
+#### Assign an image to a category
+
+Similar to the product edit scenario
+
+### Store-Front Scenarios
+
+#### Display an image on product details or products list page
+
+1. Client (GraphQL server) requests product details (including images) from the SF application.
+2. SF application returns full image URL of the original image
+
+## Synchronization from Backoffice to Store-Front
+
+### Concepts:
+
+1. Store-Front stores only original image URLs
+2. Store-Front is not responsible for physical images. This is responsibility of DAM (which can be a specialized DAM or Magento Back office). 
+2. Image URLs support image transformation by parameters (e.g., `https://some.domain.com/media/catalog/product/1/2/3.jpg?w=100&h=100`)
+   1. Image transformation is performed by CDN. The URLs passed from the Backoffice to Store-Front are CDN-based.
+   2. Image transformation can be done by web server on the Magento side (e.g., Store-Front), but this looks less efficient than using a CDN. Benefits of this approach are not clear. Such scenario is assumed useful for development scenarios only.
+3. Image types (`small`, `thumbnail`, etc) are included in the information stored on Store-Front side.
+
+
+### Questions:
+
+1. Do we sync full image URL to SF or provide Base CDN URL as configuration for the store?
+   1. Full URLs as part of product data: full reindex will be required in case CDN URL changes.
+   2. Base CDN URL as a store configuration + relative asset path as part of product data: added complexity of the SF App due to additional knowledge about CDN/Media Base URLs (especially in case multiple should be supported).
+
+## Dependencies
+
+1. To allow only original image URLs stored at the Store-Front side, new format of transformed images should be supported with transformation parameters are passed as parameters to the original URL. Current transformed URL: `https://magento.store.com/media/catalog/product/cache/1/2/3/<hash>.jpg` (where `<hash>` is generated based on transformation parameters and those become invisible). Desired transformed URL: `https://store.cdn.com/media/catalog/product/1/2/3/product-image.jpg?w=100&h=100` (transformation parameters are clearly visible).
```

Comment:
Can we add hash as parameter to desired URL format for backward compatibility purposes, like `?w=100&h=100&hash=<hash>`?
And what about other parameters like watermark?

Reply:
1. Hash. It won't give good performance I assume because currently hash is part of the file path and it's retrieved bypassing any PHP part of the application. Tough, possibly a web server can be configured to replicate same behavior. So yes, it's possible to support such format.
2. Other parameters. Would be similar. The parameters would be part of the client application based on which CDN is used in the application. I don't think Magento's Catalog responsibility (or any backend) would be to provide URL format. Drawback would be that client should know format of the transformation URL based on underlying technologies. So far, at least some CDNs can support same format: Fastly (has its own format), CloudFront relies on custom Lambda anyways (so it can support same Fastly format), Nginx/Apache  is configurable (so can also support same format as Fastly).
   1. Watermark. Yes, this part is the most questionable (mostly from UX side). Image transformation service should support image image overlay.

Will add more details about different types of image transformation.